wafer_space/projects/tasks.py

```
@@ -90,53 +90,448 @@ def _extract_filename_from_url(url: str) -> str:
+    if not check.project_file:
+        msg = "ManufacturabilityCheck must have a project_file"
+        raise ValueError(msg)
+
+    if not check.project_file.file:
+        msg = "ProjectFile has no uploaded file"
+        raise ValueError(msg)
+
+    return check.project_file
+
+
+def _handle_retry(check, error_summary, task_instance, logger):
```

Ruff (0.14.5)

Unused function argument: `task_instance`

(ARG001)